src/libraries/Common/src/System/Security/Cryptography/CompositeMLDsa.cs

```diff
@@ -0,0 +1,1800 @@
+
+            return ImportPkcs8PrivateKey(new ReadOnlySpan<byte>(source));
+        }
+
+        /// <summary>
+        ///   Imports a Composite ML-DSA private key from a PKCS#8 PrivateKeyInfo structure.
+        /// </summary>
+        /// <param name="source">
+        ///   The bytes of a PKCS#8 PrivateKeyInfo structure in the ASN.1-BER encoding.
+        /// </param>
+        /// <returns>
+        ///   The imported key.
+        /// </returns>
+        /// <exception cref="CryptographicException">
+        ///   <para>
+        ///     The contents of <paramref name="source"/> do not represent an ASN.1-BER-encoded PKCS#8 PrivateKeyInfo structure.
+        ///   </para>
+        ///   <para>-or-</para>
+        ///   <para>
+        ///     The PrivateKeyInfo value does not represent a Composite ML-DSA key.
+        ///   </para>
+        ///   <para>-or-</para>
+        ///   <para>
+        ///     <paramref name="source" /> contains trailing data after the ASN.1 structure.
+        ///   </para>
+        ///   <para>-or-</para>
+        ///   <para>
+        ///     The algorithm-specific import failed.
+        ///   </para>
+        /// </exception>
+        /// <exception cref="PlatformNotSupportedException">
+        ///   The platform does not support Composite ML-DSA. Callers can use the <see cref="IsSupported" /> property
+        ///   to determine if the platform supports Composite ML-DSA.
+        /// </exception>
+        public static CompositeMLDsa ImportPkcs8PrivateKey(ReadOnlySpan<byte> source)
+        {
+            ThrowIfInvalidLength(source);
+            ThrowIfNotSupported();
+
+            KeyFormatHelper.ReadPkcs8(s_knownOids, source, PrivateKeyReader, out int read, out CompositeMLDsa dsa);
+            Debug.Assert(read == source.Length);
+            return dsa;
+
+            static void PrivateKeyReader(
+                ReadOnlyMemory<byte> privateKeyContents,
+                in AlgorithmIdentifierAsn algorithmIdentifier,
+                out CompositeMLDsa dsa)
+            {
+                CompositeMLDsaAlgorithm algorithm = GetAlgorithmIdentifier(in algorithmIdentifier);
+                AsnValueReader reader = new AsnValueReader(privateKeyContents.Span, AsnEncodingRules.BER);
+
+                if (!reader.TryReadPrimitiveOctetString(out ReadOnlySpan<byte> key) || reader.HasData)
+                {
+                    throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                }
+
+                if (key.Length < algorithm.MLDsaAlgorithm.PrivateSeedSizeInBytes)
+                {
+                    throw new CryptographicException(SR.Argument_PrivateKeyTooShortForAlgorithm);
+                }
+
+                dsa = CompositeMLDsaImplementation.ImportPrivateKeyImpl(algorithm, key);
+            }
+        }
+
+        /// <summary>
+        ///   Imports a Composite ML-DSA public key from its private seed value.
```

Comment:
Composites don't, to my recollection, have seeds.

And, also, this is the public key import :smile: